### PR TITLE
[all_tests][python] Remove calling of get_event_loop from python version >= 3.10 

### LINKED ISSUE #25025: [Core | Python3.10] Stop using `get_event_loop`
### What happened + What you expected to happen

`get_event_loop` will be deprecated and print spammy logs to the driver. We should use `new_event_loop ` and `get_running_loop` instead. https://docs.python.org/3.10/library/asyncio-eventloop.html#asyncio.get_event_loop

### Versions / Dependencies

Ray for Python 3.10

### Reproduction script

pytest -vs python/ray/tests/test_basic.py::test_put_get

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

`get_event_loop` is deprecated in 3.10. This PR removes its invocation with python >= 3.10 by introducing a proxy function 
`get_or_create_event_loop` in `utils.py`. More details please see the function comments. 

In the long run - we should refactor the event based code by either: 
1. using `asyncio.run` as much as possible after deprecating python 3.6
2. creating and managing the event loops explicitly across threads and different contexts. 

This PR only serves as a mitigation for not calling the deprecated function. 

## Related issue number

Closes #25025

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rickyyx** [comment]
Waiting for test runs.

**Comment by rickyyx** [comment]
As a follow up - I am also planning to announce in OSS to not use `get_event_loop` in new code.

File: python/ray/_private/utils.py
Comment (jjyao) [inline python/ray/_private/utils.py:1588]: We don't support python 2.x

File: python/ray/_private/utils.py
Comment (jjyao) [inline python/ray/_private/utils.py:1592]: Should we specialize 3.6 instead of 3.10 given we will remove support for 3.6? Like calling `asyncio.get_event_loop_policy().get_event_loop()` if >=3.7?

**Comment by scv119** [comment]
let's also add a test to test_minimal https://github.com/ray-project/ray/blob/master/ci/ci.sh#L759 so it will be tested agains all python version.

File: python/ray/_private/utils.py
Comment (scv119) [inline python/ray/_private/utils.py:1576]: could this function be simplified to:

```
if vers_info.major >= 3 and vers_info.minor >= 10:
   return asyncio.get_event_loop_policy().get_event_loop()
else:
  return asyncio.get_running_loop()
```

File: python/ray/tests/BUILD
Comment (rickyyx) [inline python/ray/tests/BUILD:177]: reodering tests.

Comment (rickyyx) [reply python/ray/tests/BUILD:177]: and add test_utils.py

**Comment by rickyyx** [comment]
Triggering full CI. And ready for reviews

**Review by scv119 — APPROVED** [review]

**Comment by scv119** [comment]
need to address linter and CI failures

**Comment by rickyyx** [comment]
@scv119 Tests unrelated - should be good to merge if not.

**Review by scv119 — APPROVED** [review]

**Comment by scv119** [comment]
looks the master is almost green now; consider merge master again?

**Review by rkooo567 — APPROVED** [review]

File: python/ray/_private/utils.py
Comment (cadedaniel) [inline python/ray/_private/utils.py:1587]: Should we check the error message string? This could catch other unrelated issues, if the error is unrelated then we raise the original exception.

Comment (rickyyx) [reply python/ray/_private/utils.py]: added an assert there.

**Comment by rickyyx** [comment]
Are we gonna merge it? Or we will postpone this til later? @scv119

**Comment by scv119** [comment]
merge master?

**Comment by scv119** [comment]
test_object_metrics are failing on this PR but not master, might be related?

**Comment by rickyyx** [comment]
Oh i think that was a revert. Let me rebase.

**Review by architkulkarni — APPROVED** [review]
Serve `doc_code` changes looks good to me (codeowner)

**Review by stephanie-wang — APPROVED** [review]

**Comment by stephanie-wang** [comment]
Lint failure looks like it might be related?

**Pull request merged**
